### PR TITLE
test(sqllab): Convert tests to RTL for SqlEditor

### PR DESCRIPTION
**Pull request by corbinrobb**
### SUMMARY
Main goal of this PR is to convert `SqlEditor`'s Enzyme tests to React Testing Library with a secondary goal of removing act errors.

Most tests were checking if a component rendered or not. So the changes made were to render with RTL and check if a visual element that the user would only see if the component were or weren't there, instead of checking whether the component itself exists.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
N/A

### TESTING INSTRUCTIONS
- cd into `superset-frontend`
- `npm run test -- src/SqlLab/components/SqlEditor/SqlEditor.test.jsx`
- Check for passing tests and act errors

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/SqlLab/components/SqlEditor/SqlEditor.test.jsx [hunk 1]
```diff
@@ -263,8 +262,8 @@ describe('SqlEditor', () => {
   it('render a Limit Dropdown', async () => {
     const defaultQueryLimit = 101;
     const updatedProps = { ...mockedProps, defaultQueryLimit };
-    const wrapper = buildWrapper(updatedProps);
-    await waitForComponentToPaint(wrapper);
-    expect(wrapper.find(AntdDropdown)).toExist();
+    const { findByText } = setup(updatedProps, store);
+    fireEvent.click(await findByText('LIMIT:'));
```
Comment (EugeneTorap) [inline superset-frontend/src/SqlLab/components/SqlEditor/SqlEditor.test.jsx:266]: > Most projects have a few use cases for fireEvent, but the majority of the time you should probably use [@testing-library/user-event](https://testing-library.com/docs/user-event/intro).

Should we use `user-event` here?

**Review by EugeneTorap — COMMENTED** [review]
Hi @corbinrobb! Thank you for this nice PR!
Can we remove `import { mount } from 'enzyme';` on 20 line?

**Comment by corbinrobb** [comment]
Hey! @EugeneTorap Thank you and thanks for the review!

I would like to remove the `mount` import on line 20 and also the `buildWrapper` function on line 120 but they are still being used inside of the two skipped tests lines 235 - 260. I am not sure what is going one with those two or whether we want to keep them or not.

I notice there is a comment by @eschutho to convert them to RTL. What do you all think? Should I try to convert those two, delete them, or leave them alone?

File: superset-frontend/src/SqlLab/components/SqlEditor/SqlEditor.test.jsx (hunk 1 above)
Comment (corbinrobb) [reply superset-frontend/src/SqlLab/components/SqlEditor/SqlEditor.test.jsx:266]: I think we should!  I will switch that over and push up the changes.

**Comment by EugeneTorap** [comment]
Can we try to convert them to RTL?
@michael-s-molina @kgabryje What do you think about it?

**Comment by michael-s-molina** [comment]
> Can we try to convert them to RTL? @michael-s-molina @kgabryje What do you think about it?

I think we can try to convert them to RTL but I'm not sure if it will be easy because they are testing overflowing behavior. If it becomes too hard, then we can remove them from the file and write those tests using Cypress.

**Comment by EugeneTorap** [comment]
Hi @corbinrobb!
Can we merge your PR or you're working on rewriting skipped tests?

**Comment by corbinrobb** [comment]
Hey @EugeneTorap! I was planning on working on it this week but i have been pretty busy.

I am okay with merging it if you think that is best! I can open another PR if I am able to figure out how to convert those.

**Review by michael-s-molina — APPROVED** [review]

**Pull request merged**